File: leetcode/019_removenthnodefromendoflist.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        if head == None or head.next == None:
            return None
        
        target = ListNode(0)
        target.next = head
        end = target
        mid = target
        
        for i in range(n):
            end = end.next
        
        print(end.val)
        while end.next != None:
            print(str(end.val) + " " + str(mid.val))
            end = end.next
            mid = mid.next
            
        mid.next = mid.next.next
        
        return target.next

File: leetcode/019_removenthnodefromendoflist.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function removeNthFromEnd(head: ListNode | null, n: number): ListNode | null {
    let h = head
    let t = head
    for (let i = 0; i < n; i++) {
        h = h.next
        console.log("h", h ? h.val: "null")
    }
    if (!h) return head.next

    while (h.next != null) {
        h = h.next
        t = t.next
        console.log({h: h ? h.val: "null", t: t ? t.val : "null"})
    }

    t.next = t.next.next
    return head
};
